### PR TITLE
PackageNameFilter for including and excluding package names

### PR DESCRIPTION
**Pull request by bechte**
## Overview

- Solves #313 

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

File: junit-platform-gradle-plugin/src/main/groovy/org/junit/platform/gradle/plugin/PackagesExtension.groovy [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Package configuration options for the JUnit Platform Gradle plugin.
+ *
+ * @since 1.0
+ */
+class PackagesExtension {
+
+	/**
+	 * A list of <em>excludedPackages</em> to be included when building the test plan.
```
Comment (sbrannen) [inline junit-platform-gradle-plugin/src/main/groovy/org/junit/platform/gradle/plugin/PackagesExtension.groovy]: should say `includedPackages`.

File: junit-platform-gradle-plugin/src/main/groovy/org/junit/platform/gradle/plugin/PackagesExtension.groovy [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	 */
+	List<String> include = []
+
+	/**
+	 * A list of <em>excludedPackages</em> to be excluded when building the test plan.
+	 */
+	List<String> exclude = []
+
+	/**
+	 * Add one or more <em>excludedPackages</em> to be included when building the test plan.
```
Comment (sbrannen) [inline junit-platform-gradle-plugin/src/main/groovy/org/junit/platform/gradle/plugin/PackagesExtension.groovy]: should say `includedPackages`.

File: junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/IncludePackageNameFilter.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import org.junit.platform.commons.util.Preconditions;
+import org.junit.platform.engine.FilterResult;
+
+/**
+ * {@link PackageNameFilter} that matches fully qualified package names that
+ * are prefixed by one of the package names provided to the filter.
+ * <p>
```
Comment (marcphilipp) [inline junit-platform-engine/src/main/java/org/junit/platform/engine/discovery/IncludePackageNameFilter.java]: This `<p>` should be removed.

**Comment by bechte** [comment]
Added additional docs and corrected the minor issues mentioned above.

**Comment by bechte** [comment]
Currently, we are experiencing problems with travis. Team discussion: Merge this branch into master.

**Pull request merged**